app/hooks/map/index.ts

```diff
@@ -435,6 +528,7 @@ export function usePUGridLayer({
             {
               type: 'fill',
               'source-layer': 'layer0',
+              visibility: SolutionVisibility,
```

Comment:
Why you don't apply the same function you applied to the rest of layers `getLayerVisibility `?